lib/sqlalchemy/sql/elements.py

```diff
@@ -1906,6 +1921,18 @@ def and_(cls, *clauses):
             :func:`.or_`
 
         """
+        if NONE_CLAUSE_HANDLING in ("warning", "error") or NONE_CLAUSE_HANDLING.startswith("log:"):
+            if not isinstance(clauses, (list, tuple)):
+                clauses = list(clauses)
+            if type(None) in map(type, clauses):
+                if NONE_CLAUSE_HANDLING == "warning":
+                    warnings.warn("Use of None in and_", NoneClauseWarning)
+                    clauses = [clause for clause in clauses if clause is not None]
```

Comment:
So we need to set it to 'warning' to get it to fix the stuff? log:warn will behave like normal 1.0.6 except for logging extra info?